app/filtering.js

```diff
@@ -495,9 +496,10 @@ function registerPermissionHandler (session, partition) {
         }
       }
 
+      let parsedMainFrameUrl = new URL(mainFrameUrl)
```

Comment:
can this use `urlParse`?

Reply:
With `urlParse`:

```
tps://oogle.com/search?q=s%3A%2F%2Fwww.lcebook.com%2F
{ hash: '',
  hostname: '',
  host: '',
  href: 'tps://oogle.com/search?q=s%3A%2F%2Fwww.lcebook.com%2F',
  path: '//oogle.com/search?q=s%3A%2F%2Fwww.lcebook.com%2F',
  pathname: '//oogle.com/search',
  port: '',
  protocol: 'tps:',
  query: 'q=s%3A%2F%2Fwww.lcebook.com%2F',
  search: '?q=s%3A%2F%2Fwww.lcebook.com%2F',
  origin: '' }
```

Reply:
with `new URL`
```
URL {
  href: tps://oogle.com/search?q=s%3A%2F%2Fwww.lcebook.com%2F
  protocol: tps:
  hostname: oogle.com
  pathname: /search
  search: ?q=s%3A%2F%2Fwww.lcebook.com%2F
}
```

Reply:
Fails here: `<a href="tp://ttt.com">Test Url</a>`